service/src/main/java/org/cbioportal/service/impl/SampleServiceImpl.java

```diff
@@ -174,31 +174,51 @@ public BaseMeta fetchMetaSamples(List<String> sampleListIds) {
         return sampleRepository.getSamplesByInternalIds(internalIds);
     }
 
-    private void processSamples(List<Sample> samples, String projection) {
+    private Boolean isProfiledWithSV(MolecularProfile p) {
+        return (p.getMolecularAlterationType().equals(MolecularProfile.MolecularAlterationType.STRUCTURAL_VARIANT) ||
+                hasFusionsAsMutations(p));
+    }
 
+    private Boolean hasFusionsAsMutations(MolecularProfile p) {
+        return (p.getMolecularAlterationType().equals(MolecularProfile.MolecularAlterationType.MUTATION_EXTENDED) &&
+                p.getDatatype().equals("FUSION"));
+    }
+
+    private void processSamples(List<Sample> samples, String projection) {
         if (projection.equals("DETAILED")) {
             Map<String, Set<String>> sequencedSampleIdsMap = new HashMap<>();
             Map<String, Set<String>> structuralVariantSampleIdsMap = new HashMap<>();
             List<String> distinctStudyIds = samples.stream().map(Sample::getCancerStudyIdentifier).distinct()
                 .collect(Collectors.toList());
             List<MolecularProfile> molecularProfiles = molecularProfileRepository.getMolecularProfilesInStudies(distinctStudyIds, projection);
             List<String> studiesProfiledWithSVs = molecularProfiles.stream()
-                        .filter(p -> p.getMolecularAlterationType().equals(MolecularProfile.MolecularAlterationType.STRUCTURAL_VARIANT))
+                        .filter(p -> isProfiledWithSV(p))
+                        .map(MolecularProfile::getCancerStudyIdentifier)
+                        .collect(Collectors.toList());
+            List<String> studiesProfiledWithFusionsAsMutations = molecularProfiles.stream()
+                        .filter(p -> hasFusionsAsMutations(p))
                         .map(MolecularProfile::getCancerStudyIdentifier)
                         .collect(Collectors.toList());
             for (String studyId : distinctStudyIds) {
                 sequencedSampleIdsMap.put(studyId,
                                           new HashSet<String>(sampleListRepository.getAllSampleIdsInSampleList(studyId + SEQUENCED)));
-                structuralVariantSampleIdsMap.put(studyId,
-                                       new HashSet<String>(sampleListRepository.getAllSampleIdsInSampleList(studyId + STRUCTURAL_VARIANT)));
+                // get sv samples from sequenced case list if fusions imported as mutations
+                Set<String> svSamples = new HashSet<>();
+                if (studiesProfiledWithFusionsAsMutations.contains(studyId)) {
+                    // use sample list that has already been fetched
+                    svSamples = new HashSet<>(sequencedSampleIdsMap.get(studyId));
+                } else {
+                    svSamples = new HashSet<String>(sampleListRepository.getAllSampleIdsInSampleList(studyId + STRUCTURAL_VARIANT));
```

Comment:
Is this hit when the profile alteration type is STRUCTURAL_VARIANT?

Reply:
yes